### PR TITLE
dobi: node v10

### PR DESCRIPTION
**Pull request by SJAnderson**
### To Test:

1. in bash_rc set an alias like: 
alias mydobi='coffee ~/dev/lessthan3/dobi/lib/cli.coffee'
2. CD into your dobi repository
3. pull branch "upgrade" in dobi, 
5. nvm use 8.9
4. npm install
5. CD into your packages directory
6. nvm use 8.9
7. mydobi run

### Dobi is our middleware and dev tool for live development.

Changes - Dependencies:
- Async to 2.6 from 0.9
- dobi-asset-wrap from asset-wrap
- chokidar to 2.0 from 0.11
- coffeescript instead of coffee-script
- colors to 1.1 from 0.6
- columnify to 1.5 from 1.2
- copy-paste to 1.3 from 0.3
- dobi-mongofb instead of mongofb
- firebase to 2.2 from 1.0
- ncp to 2.0 from 0.5
- node.extend to 2.0 from 1.0
- request to 2.83 from 2.39
- season to 6.0.2 from 5.3.0
- underscore to 1.8 from 1.5
- upgrade node engine to above 6.11

Changes - server.coffee (will affect rooms):
- upgrading asset-wrap to dobi-asset-wrap
- only logging `read` logs if in dev mode
- turning console.logs on errors to console.errors

Changes - cli.coffee (our command line tool):
- updating the DATABASE_URL to be current
- using minimalist instead of optimist since that's been deprecated
- fixing `dobi cache:warm` to work (we never use it, however)

### CODE REVIEW

File: lib/cli.coffee [hunk 1]
```diff
@@ -346,17 +345,18 @@ switch command
 
       # query DB for domains of site slugs
       getSiteDomains = (sites, done) ->
-        SLUGS = for site in sites
```
Comment (AndyBarron) [inline lib/cli.coffee]: kinda liked the old way better tbh. or you could do a `map` and then a `filter (slug) -> slug`. not a fan of coffeescript loop magic :eyes:

File: package.json [hunk 2]
```diff
@@ -58,5 +60,5 @@
     "type": "git",
     "url": "git://github.com/lessthan3/dobi.git"
   },
-  "version": "2.0.10"
+  "version": "2.1.0-beta"
```
Comment (AndyBarron) [inline package.json]: `2.1.0-beta.1`

File: package.json [hunk 3]
```diff
@@ -15,40 +15,42 @@
...
-    "underscore": "1.5.x"
+    "underscore": "1.8.x"
   },
   "description": "LessThan3 Developer Tools",
   "directories": {
     "lib": "./lib",
     "bin": "./bin"
   },
+  "engines": {
+    "node": ">=6.11.x"
```
Comment (AndyBarron) [inline package.json]: 😃 😄 😁

File: package.json [hunk 4]
```diff
@@ -15,40 +15,42 @@
...
+    "columnify": "1.5.4",
     "connect": "2.17.x",
-    "copy-paste": "0.3.x",
+    "copy-paste": "1.3.x",
+    "dobi-asset-wrap": "2.0.2",
+    "dobi-mongofb": "1.3.0",
     "express": "3.4.x",
     "findit": "1.2.x",
-    "firebase": "1.0.x",
+    "firebase": "2.2.9",
```
Comment (AndyBarron) [inline package.json]: you can bump this up to the very last 2.x release (i think `2.4.2`) - just adds some bug fixes and promises (but still supports callbacks)

File: lib/server.coffee [hunk 5]
```diff
@@ -448,7 +449,7 @@ exports = module.exports = (cfg) ->
       file = file.join path.sep
       console.log "#{id} v#{version} updated"
       readConfig id, version, (err, config) ->
-        return console.log(err) if err
+        return console.error(err) if err
```
Comment (kamronbatman) [inline lib/server.coffee]: Unnecessary paren

File: lib/server.coffee [hunk 6]
```diff
@@ -463,7 +464,7 @@ exports = module.exports = (cfg) ->
           file_ext: path.extname(file).replace '.', ''
           file_name: path.basename file, path.extname(file)
         ref.set config, (err) ->
-          console.log(err) if err
+          console.error(err) if err
```
Comment (kamronbatman) [inline lib/server.coffee]: Same

File: package.json (hunk 2 above)
Comment (kamronbatman) [inline package.json]: This is a breaking change and should be major semver. I vote v3.0.0-beta-1

File: lib/cli.coffee (hunk 1 above)
Comment (SJAnderson) [reply lib/cli.coffee]: addressed in following line

File: lib/server/index.js [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+  cache: defaultCache,
+  error,
+  getPackage,
+  readConfig,
+  readSchema,
+  setContentType,
+} = require('./helpers');
+
+const asyncExists = promisify(fs.exists);
+const firebase = new Firebase('https://lessthan3.firebaseio.com');
```
Comment (SJAnderson) [inline lib/server/index.js]: should data drive this.

**Comment by SJAnderson** [comment]
not merging in favor of whats in master with koa

**Pull request closed without merging**